source/Ship.cpp

```diff
@@ -404,6 +404,17 @@ void Ship::Load(const DataNode &node)
 				else
 					grand.PrintTrace("Skipping invalid outfit count:");
 			}
+			
+			// Verify we have at least as many installed outfits as were identified as "equipped."
+			// If not (e.g. a variant definition), ensure FinishLoading equips into a blank slate.
+			if(!hasArmament)
+				for(const auto &it : equipped)
+					if(static_cast<int>(outfits.count(it.first)) < it.second)
```

Comment:
.... I'm dumb. @tehhowch this code has an error that I didn't notice (as usual). Sometimes I'm blind when it comes to errors in my code. 

`outfits` is a map so `count` returns either `1` or `0.... It should `find` and then use the value of the pair to compare.

Reply:
Yeah, I missed it too when checking the bugs that were fixed by the new logic. I had even taken notice of the need for the static cast due to the type difference, but missed that it was due to comparing the wrong things. Knowing what you were intending to do, plus the wonderful language semantics (_Of course `outfits.count` counts the outfits, that's the point of the condition comparison..._), was just enough to get us both. 😅